pkg/ingestor/parser/csaf/parser_csaf.go

```diff
@@ -0,0 +1,252 @@
+	for _, t := range tree.Threats {
+		if t.Category == "impact" {
+			for _, p := range t.ProductIDs {
+				if p == product_id {
+					return &t.Details
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func (c *csafParser) findPkgSpec(ctx context.Context, product_id string) (*generated.PkgInputSpec, error) {
+	pref := findProductRef(ctx, c.csaf.ProductTree, product_id)
+	if pref == nil {
+		return nil, fmt.Errorf("unable to locate product reference for id %s", product_id)
+	}
+
+	purl := findPurl(ctx, c.csaf.ProductTree, *pref)
+	if purl == nil {
+		return nil, fmt.Errorf("unable to locate product url for reference %s", *pref)
+	}
+
+	return helpers.PurlToPkg(*purl)
+}
+
+func (c *csafParser) generateVexIngest(ctx context.Context, cve *generated.CVEInputSpec, vuln *csaf.Vulnerability, status string, product_id string) *assembler.VexIngest {
+	logger := logging.FromContext(ctx)
+	vi := &assembler.VexIngest{}
+
+	vd := generated.VexStatementInputSpec{}
+	vd.KnownSince = c.csaf.Document.Tracking.CurrentReleaseDate
+	vd.Origin = c.csaf.Document.Tracking.ID
+	vd.VexJustification = generated.VexJustificationNotProvided
+
+	if vexStatus, ok := vexStatusMap[status]; ok {
+		vd.Status = vexStatus
+	}
+
+	var statement *string
+	if vd.Status == generated.VexStatusNotAffected {
+		statement = findImpactStatement(vuln, product_id)
+	} else {
+		statement = findActionStatement(vuln, product_id)
+	}
+
+	if statement != nil {
+		vd.Statement = *statement
+	}
+
+	for _, flag := range vuln.Flags {
+		found := false
+		for _, pid := range flag.ProductIDs {
+			if pid == product_id {
+				found = true
+			}
+		}
+		if found {
+			if just, ok := justificationsMap[flag.Label]; ok {
+				vd.VexJustification = just
+			}
+		}
+	}
+
+	vi.VexData = &vd
+	vi.CVE = cve
+
+	pkg, err := c.findPkgSpec(ctx, product_id)
+	if err != nil {
+		logger.Warnf("[csaf] unable to locate package for not-affected product %s", product_id)
+		return nil
+	}
+	vi.Pkg = pkg
+
+	return vi
+}
+
+func (c *csafParser) GetPredicates(ctx context.Context) *assembler.IngestPredicates {
+	rv := &assembler.IngestPredicates{}
+	var vis []assembler.VexIngest
+	var cvs []assembler.CertifyVulnIngest
+
+	if len(c.csaf.Vulnerabilities) > 0 {
+
+		for _, v := range c.csaf.Vulnerabilities {
+			// CVE field only contains CVE id, use the helper to parse and create the CVE input spec.
+			cve, _, err := helpers.OSVToGHSACVE(v.CVE)
+			if err != nil || cve == nil {
+				return nil
+			}
+
+			statuses := []string{"fixed", "known_not_affected", "known_affected", "first_affected", "first_fixed", "last_affected", "recommended", "under_investigation"}
+			for _, status := range statuses {
+				products := v.ProductStatus[status]
+				if len(products) > 0 {
+					for _, product := range products {
+						vi := c.generateVexIngest(ctx, cve, &v, status, product)
+						if vi == nil {
+							continue
+						}
+
+						if status == "known_affected" || status == "under_investigation" {
```

Comment:
if not affected, we can also add CertifyVulnIngest with `NoVuln` as the target (https://github.com/guacsec/guac/blob/main/pkg/assembler/graphql/schema/certifyVuln.graphql#L63)

Reply:
Issue added to track this.